FLEDGE.md

```diff
@@ -136,14 +148,20 @@ The `dailyUpdateUrl` provides a mechanism for the group's owner to periodically
 
-The `ads` list contains the various ads that the interest group might show.  Each entry is an object that includes both a rendering URL and arbitrary metadata that can be used at bidding time.
+The `ads` list contains the various ads that the interest group might show.  Each entry is an object that includes a rendering URL, a named size group (see below), and arbitrary metadata that can be used at bidding time. These render URLs may contain macros `{%AD_WIDTH%}` and `{%AD_HEIGHT%}`, which will be automatically replaced with the appropriate width and height after an auction, so that the initial resource request can fetch appropriately sized assets.
 
-The `adComponents` field contains the various ad components (or "products") that can be used to construct ["Ads Composed of Multiple Pieces"](https://github.com/WICG/turtledove/blob/main/FLEDGE.md#34-ads-composed-of-multiple-pieces)). Similarly to the `ads` field, each entry is an object that includes both a rendering URL and arbitrary metadata that can be used at bidding time. Thanks to `ads` and `adsComponents` being separate fields, the buyer is able to update the `ads` field via daily update without losing `adComponents` stored in the interest group.
+The `adComponents` field contains the various ad components (or "products") that can be used to construct ["Ads Composed of Multiple Pieces"](https://github.com/WICG/turtledove/blob/main/FLEDGE.md#34-ads-composed-of-multiple-pieces)). Similarly to the `ads` field, each entry is an object that includes a rendering URL, a named size group (see below), and arbitrary metadata that can be used at bidding time. Thanks to `ads` and `adsComponents` being separate fields, the buyer is able to update the `ads` field via daily update without losing `adComponents` stored in the interest group.
+
+The `adSizes` field contains a dictionary of named ad sizes. Each size has the format `{width: widthVal, height: heightVal}`, where the values can have either pixel units (e.g. `100` or `'100px'`) or screen dimension coordinates (e.g. `100sw` or `100sh`). For example, the size `{width: '100sw', height: 50}` describes an ad that is the width of the screen and 50 pixels tall. The size `{width: '100sw', height: '200sw'}` describes an ad that is the width of the screen and has a 1:2 aspect ratio. Sizes with screen dimension coordinates are primarily intended for screen-width ads on mobile devices, and may be restricted in certain contexts (to be determined) for privacy reasons.
```

Comment:
In the current Chrome implementation all sizes have to have units ('px', 'sq', or 'sh'). Is it intended to allow unitless numbers as a shortcut for "px"?

Reply:
Does [our implementation](https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:third_party/blink/public/mojom/interest_group/ad_display_size.mojom;drc=86603c7109c51c7cac2b3f6e473c4473f321a1eb;l=14) support "sh" as a valid unit? It seems only "sw" and "px" are supported. Am I missing something?

Reply:
@brusshamilton It is intended to allow unitless numbers as a shortcut for "px", but currently doesn't.
@qingxinwu Lack of "sh" is also a known issue.

We will fix both of these problems imminently (i.e. for M114), thanks.

Reply:
Possibly a silly question, but where are `sw` or `sh` defined? I don't see them in https://drafts.csswg.org/css-values-4/#viewport-relative-units (are they some magic ads thing)?

Reply:
Not silly at all: they aren't CSS units. As far as we can tell (I consulted with some screen/layout experts), there are no existing units relative to the screen dimensions, rather than window etc. dimensions (which don't satisfy the privacy properties we need, namely that they can't be modified by the web platform).

This is nice in that we aren't beholden to some existing behavior specification; we can make the behavior exactly what is necessary for FLEDGE privacy/utility.

Reply:
Ah, interesting - thanks! I assume we'll define those at some later point in the spec.